src/event-stream/tokens-contract-handler.ts

```
@@ -0,0 +1,417 @@
+      }
+    } catch (error) {
+      logger.error('erro: error handling NFT contract', error);
+    }
+  }
+
+  /**helpng method for creating http url */
+  private makeHostedUrl(uri: string): string {
+    if (uri.includes('http')) return uri;
+    const parsedURI = URI.parse(uri);
+    return `${PUBLIC_IPFS}/${parsedURI.host}${parsedURI.path}`;
+  }
+
+  async makeApiCall<Type>(url: string): Promise<Type> {
+    const result = await fetch(url);
+    if (!result.ok) {
+      let msg = '';
+      try {
+        msg = await result.text();
+      } catch (error) {
+        logError(`Error getting text`, error);
+      }
+      throw new Error(`Response ${result.status}: ${result.statusText} fetching ${url} - ${msg}`);
+    }
+    try {
+      const resultString = await result.text();
+      return JSON.parse(resultString) as Type;
+    } catch (error) {
+      logError(`Error reading response from ${url}`, error);
+      throw error;
+    }
+  }
+  /**
+   * fetch metadata from uri
+   */
+  private async getMetadataFromUri<Type>(token_uri: string): Promise<Type> {
+    return await this.makeApiCall(this.makeHostedUrl(token_uri));
+  }
+
+  /**
+   * make readonly contract call
+   */
+
+  private async makeReadOnlyContractCall(
+    functionName: string,
+    functionArgs: ClarityValue[]
+  ): Promise<ClarityValue> {
+    const txOptions: ReadOnlyFunctionOptions = {
+      senderAddress: this.address,
+      contractAddress: this.contractAddress,
+      contractName: this.contractName,
+      functionName: functionName,
+      functionArgs: functionArgs,
+      network: this.stacksNetwork,
+    };
+    return await callReadOnlyFunction(txOptions);
+  }
+
+  /**
+   *Store ft metadata to db
+   */
+  private async storeFTMetadata(ft_metadata: DbFungibleTokenMetadata) {
+    try {
+      await this.db.updateFtMetadata(ft_metadata);
+    } catch (error) {
+      logger.error(`Error: error occured while updating FT metadata`);
```

Should throw here otherwise error is swallowed rather than handled more appropriately in a function higher up in the call graph.